## Función `velocidad`

La función `velocidad` calcula la velocidad en dos unidades de medida: kilómetros por hora (km/h) y metros por segundo (m/s), dados la distancia recorrida y el tiempo empleado.

### Propósito

Esta función permite obtener la velocidad de un recorrido en las unidades mencionadas, realizando las conversiones necesarias y proporcionando el resultado en una cadena de texto sin redondeo, mostrando todos los decimales.

### Parámetros

 `distancia` (float): La distancia recorrida en kilómetros. Debe ser mayor o igual a 0.

 `tiempo` (float): El tiempo empleado en segundos. Debe ser mayor a 0.

### Retorno

La función retorna un `str` que contiene el mensaje con la velocidad en **km/h** y **m/s**, sin redondeo, mostrando todos los decimales de los cálculos.

### Validación de entrada

 La función valida que la **distancia** no sea negativa.

 La función valida que el **tiempo** sea mayor que 0.

Si alguna de estas condiciones no se cumple, se lanza un error `ValueError` con un mensaje adecuado.

### Fórmulas utilizadas

1. Para convertir la velocidad a kilómetros por hora (km/h), se usa la fórmula:
`velocidad_kmh = (distancia / tiempo) * 3600`

  Donde `3600` es el número de segundos en una hora.

2. Para convertir la velocidad a metros por segundo (m/s), se usa la fórmula:
`velocidad_ms = (distancia * 1000) / tiempo`

  Donde `1000` convierte los kilómetros a metros.








In [4]:
def velocidad(distancia, tiempo):
    """
    Calcula la velocidad en kilómetros por hora y en metros por segundo.

    Parámetros:
        distancia (float): La distancia recorrida en kilómetros (debe ser >= 0).
        tiempo (float): El tiempo empleado en segundos (debe ser > 0).

    Retorna:
        str: Un mensaje con la velocidad en km/h y m/s con todos los decimales.
    """
    # Validación de entrada
    if distancia < 0:
        raise ValueError("La distancia no puede ser negativa.")
    if tiempo <= 0:
        raise ValueError("El tiempo debe ser mayor a 0.")

    # Calcular la velocidad
    velocidad_kmh = (distancia / tiempo) * 3600  # Conversión a km/h
    velocidad_ms = (distancia * 1000) / tiempo  # Conversión a m/s

    # Formatear el resultado sin redondear
    resultado = f"La velocidad es {velocidad_kmh} km/h o {velocidad_ms} m/s"

    return resultado

# Ejemplo de uso
if __name__ == "__main__":
    distancia = 66.66915082039476 # Distancia en kilómetros
    tiempo = 88.68885498283579  # Tiempo en segundos

    try:
        resultado = velocidad(distancia, tiempo)
        print(resultado)
    except ValueError as e:
        print(f"Error: {e}")

La velocidad es 2706.1905692645455 km/h o 751.719602573485 m/s


### **Resultado esperado**
Con los valores proporcionados en el ejemplo de uso (distancia = 66.66915082039476 km y tiempo = 88.68885498283579 s), el resultado esperado será:

La velocidad es 2706.190569 km/h o 751.719603 m/s